pandas/_libs/groupby.pyx

```diff
@@ -887,7 +889,7 @@ def group_last(rank_t[:, :] out,
             for j in range(K):
                 val = values[i, j]
 
-                if val == val:
+                if (val is not NA) and (val == val):
```

Comment:
I think can use the `checknull` function from missing instead

Reply:
Maybe just `not checknull(val)` for the whole condition? I think there was an issue with `checknull` not catching `decimal.Decimal("nan")` but I could try to add that here